src/lib/Helpers/ValidationHelpers.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Text.RegularExpressions;
+
+namespace Microsoft.OpenApi.ApiManifest.Helpers
+{
+    internal static class ValidationHelpers
+    {
+        internal static void ValidateNullOrWhitespace(string parameterName, string? value, string parentName)
+        {
+            if (string.IsNullOrWhiteSpace(value))
+                throw new ArgumentNullException(parameterName, string.Format(ErrorMessage.FieldIsRequired, parameterName, parentName));
+        }
+
+        internal static void ValidateNullObject(string parameterName, object? value, string parentName)
```

Comment:
replace this and its use by ArgumentNullException.ThrowIfNull(parameter). It captures the parameter name automatically and has a default message